### ISSUE TITLE
"Expression is a value and therefore cannot be the target of an assignment." after adding ReadOnly

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#47198]
**Steps to Reproduce**:

https://github.com/dotnet/roslyn/pull/47196

**Expected Behavior**:

No compile errors.

**Actual Behavior**:

The CI fails with:

> SyntaxNodeExtensions.vb(542,25): error BC30068: Expression is a value and therefore cannot be the target of an assignment.

Note: If this is not a compiler bug, then it should be an IDE0044 bug. But I don't see any reason that ReadOnly isn't allowed.

This was detected in #47148

cc: @sharwell

**Comment by Youssef1313** [comment]
@AlekseyTs Can you take a look please?

The line where the error is produced is:

https://github.com/dotnet/roslyn/blob/6982e409f0519888d094cf387151bcd1bb38bea1/src/Compilers/VisualBasic/Portable/Syntax/InternalSyntax/SyntaxNodeExtensions.vb#L542

The error was triggered after ReadOnly was added to `_triviaListBuilder` based on the suggestion of IDE0044. I don't see how `_triviaListBuilder` is assigned in this line? I'm only setting a value of the indexer.

If this compiler behavior by design, I think there is a room to improve the diagnostic here, and also IDE0044 would need to be fixed.

**Comment by AlekseyTs** [comment]
The error is correct and it is pretty clear in my opinion. The assumption here is that, if an assignment to a property is made (an indexer is a property), the intent is to modify content of the field. It is illegal to modify content of a readonly field outside of the containing type's constructor. Note, that a property assignment is translated to a setter call. All method calls on a struct readonly fields outside of the containing type's constructor are never performed directly on the field, they are performed on a local copy in order to prevent any mutations that the method can potentially cause to the field. Given the assumption that most likely the intent of the assignment is actually to modify content of the field, silently invoking the setter on a copy isn't what the user would want.

**Comment by CyrusNajmabadi** [comment]
Yup.  This is a bug in the IDE.  We need to not offer readonly here.  We're missing a case where it would change an lvalue to an rvalue.

**Comment by Youssef1313** [comment]
Aha I was missing it's a struct. Thanks @AlekseyTs and @CyrusNajmabadi .

Please switch to Area-IDE.
